packages/components/src/hooks/use-hover.ts

```diff
@@ -13,26 +14,32 @@ export const useHover = <T extends HTMLElement & SVGSVGElement>(
 
     React.useEffect(() => {
         const node = ref.current;
-        if (node) {
-            if (should_prevent_bubbling) {
-                node.addEventListener('mouseenter', handleHoverBegin);
-                node.addEventListener('mouseleave', handleHoverFinish);
+
+        if (!node) return;
+
+        const handleAddEvents = () => {
+            if (isMobileOs()) {
+                node.addEventListener('touchstart', handleHoverBegin);
+                node.addEventListener('touchend', handleHoverFinish);
+            } else {
+                node.addEventListener(should_prevent_bubbling ? 'mouseenter' : 'mouseover', handleHoverBegin);
+                node.addEventListener(should_prevent_bubbling ? 'mouseleave' : 'mouseout', handleHoverFinish);
+            }
+        };
+
+        const handleRemoveEvents = () => {
+            if (isMobileOs()) {
+                node.removeEventListener('touchstart', handleHoverBegin);
+                node.removeEventListener('touchend', handleHoverFinish);
             } else {
-                node.addEventListener('mouseover', handleHoverBegin);
-                node.addEventListener('mouseout', handleHoverFinish);
+                node.removeEventListener(should_prevent_bubbling ? 'mouseenter' : 'mouseover', handleHoverBegin);
+                node.removeEventListener(should_prevent_bubbling ? 'mouseleave' : 'mouseout', handleHoverFinish);
             }
+        };
+
+        handleAddEvents();
 
-            return () => {
-                if (should_prevent_bubbling) {
-                    node.removeEventListener('mouseenter', handleHoverBegin);
-                    node.removeEventListener('mouseleave', handleHoverFinish);
-                } else {
-                    node.removeEventListener('mouseover', handleHoverBegin);
-                    node.removeEventListener('mouseout', handleHoverFinish);
-                }
-            };
-        }
-        return undefined;
+        return handleRemoveEvents;
```

Comment:
Currently in prod, for android we dont show the label when touching the popover. With this implementation, we show the popover text on touching the icon. Is this expected change?

Reply:
In android it does show the popovers just that user have to hold for 1 second then only it will display the popovers